### PR TITLE
[Product][Shop] Buy product in correct prices for channel and currency

### PR DESCRIPTION
**Pull request by Zales0123**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | yes
| BC breaks?      | no
| Related tickets | |
| License         | MIT

### CODE REVIEW

**Comment by Zales0123** [comment]
I've opened this PR accidentally (I didn't want to open it yet) - but then I've just realized I can't lose such a beautiful PR number as #6666 😄

**Comment by Zales0123** [comment]
@pjedrzejewski @michalmarcinkowski do we want also to implement displaying product price based on pricing calculator on product's details page? Or is it the case for the separate PR?

File: src/Sylius/Behat/Context/Setup/CurrencyContext.php [hunk 1]
```diff
@@ -150,9 +150,6 @@ public function thatChannelAllowsToShopUsingAndCurrencies(ChannelInterface $chan
     public function itUsesTheCurrencyByDefault(ChannelInterface $channel, $currencyCode)
     {
         $currency = $this->provideCurrency($currencyCode);
-        $currency->setExchangeRate(1.0);
```
Comment (michalmarcinkowski) [inline src/Sylius/Behat/Context/Setup/CurrencyContext.php]: Why removing this?

File: src/Sylius/Behat/Context/Setup/ProductContext.php [hunk 2]
```diff
@@ -540,6 +558,33 @@ public function thisProductHasAnImageWithACode(ProductInterface $product, $image
     }
 
     /**
+     * @Given /^(it) has different prices for different channels and currencies$/
+     */
+    public function itHasDifferentPricesForDifferentChannelsAndCurrencies(ProductInterface $product)
+    {
+        $product->getFirstVariant()->setPricingCalculator(Calculators::CHANNEL_AND_CURRENCY_BASED);
```
Comment (michalmarcinkowski) [inline src/Sylius/Behat/Context/Setup/ProductContext.php]: `getFirstVariant` method should be removed. `defaultVariantResolver` should be used instead.

File: src/Sylius/Behat/Context/Setup/ProductContext.php [hunk 3]
```diff
@@ -540,6 +558,33 @@ public function thisProductHasAnImageWithACode(ProductInterface $product, $image
...
+    /**
+     * @Given /^(it) has price ("[^"]+") for ("[^"]+" channel) and "([^"]+)" currency$/
+     */
+    public function itHasPriceForChannelAndCurrency(
+        ProductInterface $product,
+        $price,
+        ChannelInterface $channel,
+        $currency
+    ) {
+        $variant = $product->getFirstVariant();
```
Comment (michalmarcinkowski) [inline src/Sylius/Behat/Context/Setup/ProductContext.php]: Same as above.

File: src/Sylius/Behat/Resources/config/services/contexts/setup.xml [hunk 4]
```diff
@@ -20,6 +20,7 @@
             <argument type="service" id="__symfony__.sylius.factory.channel" />
             <argument type="service" id="__symfony__.sylius.repository.channel" />
             <argument type="service" id="__symfony__.sylius.manager.channel" />
+            <argument>%__behat__.mink.parameters%</argument>
```
Comment (michalmarcinkowski) [inline src/Sylius/Behat/Resources/config/services/contexts/setup.xml]: Do we need this?

Comment (Zales0123) [inline src/Sylius/Behat/Resources/config/services/contexts/setup.xml]: No 👍  😄

File: src/Sylius/Behat/Context/Setup/CurrencyContext.php (hunk 1 above)
Comment (Zales0123) [inline src/Sylius/Behat/Context/Setup/CurrencyContext.php]: This step is ``that channel uses the currency by default`` -> it should not set this currency's exchange rate to *1.0*, as exchange rate is not configured per channel, but for whole system. In my case, it caused a mistake when one channel has base currency ``EUR`` and the other one ``GBP``.

**Comment by Zales0123** [comment]
@pamil Not anymore 🐫

File: src/Sylius/Behat/Context/Setup/CurrencyContext.php (hunk 1 above)
Comment (michalmarcinkowski) [inline src/Sylius/Behat/Context/Setup/CurrencyContext.php]: Ok, so `it` means channel, not store. Totally valid fix, good catch! 👍

**Comment by michalmarcinkowski** [comment]
Thank you Mateusz! Please apply all comments in a separate PR.

**Pull request merged**